File: 2 Array:List/7. UniqueNumber.py
def unique(arr):
    count_dict = {}
    for num in arr:
        if num in count_dict:
            count_dict[num] += 1
        else:
            count_dict[num] = 1

    for num, count in count_dict.items():
        if count == 1:
            unique_elements = num
    print(unique_elements)


arr = [1,2,3,4,1,2,3]
unique(arr)

File: 2 Array:List/7.Unique_Number.js
function unique(arr) {
    let countDict = {};
    for (let num of arr) {
        countDict[num] = (countDict[num] || 0) + 1;
    }
    let uniqueElements = [];
    for (let num in countDict) {
        if (countDict[num] === 1) {
            uniqueElements.push(Number(num));
        }
    }

    console.log(uniqueElements);
}

const arr = [1, 2, 3, 4, 1, 2, 3];
unique(arr);
